### PR TITLE
Add editorconfig to guarantee the basic code style

### LINKED ISSUE #32: Add editorconfig to guarantee the basic code style
Done. 
BTW, We can use [EditorConfig](https://editorconfig.org/) to guarantee that behavior.

_Originally posted by @yuezhuangshi in https://github.com/kubesphere/devops-agent/pull/31#discussion_r690938887_

### PR DESCRIPTION
**Pull request by yuezhuangshi**
fix #32

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+[*]
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.yml]
+indent_style = space
+indent_size = 2
+max_line_length = 120
```
Comment (LinuxSuRen) [inline .editorconfig:12]: I didn't find [any spec](https://editorconfig-specification.readthedocs.io/) of this rule. I'm wondering if you have any link which can describe it.

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]
This is great. People don't need always to pay attention to the file style. But I left a small comment, please take a look at it.

File: .editorconfig (hunk 1 above)
Comment (yuezhuangshi) [reply .editorconfig:12]: https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#max_line_length

Comment (LinuxSuRen) [reply .editorconfig:12]: I see. But it does not work well on IntelliJ IDEA.

image

```
IntelliJ IDEA 2019.2.2 (Community Edition)
Build #IC-192.6603.28, built on September 6, 2019
Runtime version: 11.0.3+12-b304.56 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.15.7
GC: ParNew, ConcurrentMarkSweep
Memory: 1979M
Cores: 12
Registry: 
Non-Bundled Plugins: Native2Ascii, com.shuqi.tools.translate-properties, native2asciiplug, Docker
```

Comment (LinuxSuRen) [reply .editorconfig:12]: See also [editorconfig: remove max_line_length directives #7761](https://github.com/MaterializeInc/materialize/pull/7761)

Comment (yuezhuangshi) [reply .editorconfig:12]: Since this rule is going to be removed, let's remove it?

Comment (LinuxSuRen) [reply .editorconfig:12]: +1 remove it

**Review by LinuxSuRen — APPROVED** [review]

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+insert_final_newline = true
+
+[*.yml]
+indent_style = space
+indent_size = 2
+
+[*.md]
+trim_trailing_whitespace = false
+
+[Makefile]
```
Comment (JohnNiang) [inline .editorconfig:16]: Please add file selector end with `.mk` here.

```editorconfig
[{Makefile,**.mk}]
```

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,17 @@
+root = true
+
+[*]
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.yml]
```
Comment (JohnNiang) [inline .editorconfig:9]: Please change to `[*.{yml,yaml}]` to support `yaml` file as well.

**Review by JohnNiang — CHANGES_REQUESTED** [review]
Please also add new editorconfig rule for `Dockerfile`.

File: .editorconfig (hunk 2 above)
Comment (yuezhuangshi) [reply .editorconfig]: Done

File: .editorconfig (hunk 3 above)
Comment (yuezhuangshi) [reply .editorconfig]: Done

**Comment by yuezhuangshi** [comment]
> Please also add new editorconfig rule for `Dockerfile` and `Jenkinsfile`. Because this repo has so many those files.

Done, `*` will affect all files, except the files that have been overwritten.

**Comment by LinuxSuRen** [comment]
Good catch from @JohnNiang

File: .editorconfig [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+[*]
+charset = utf-8
+end_of_line = lf
+indent_size = 4
+indent_style = space
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[*.{yml,yaml,sh}]
```
Comment (JohnNiang) [inline .editorconfig:11]: I suggest separating `yaml` rule and `sh` rule. Just like `Jenkinsfile`.

**Review by JohnNiang — CHANGES_REQUESTED** [review]

File: .editorconfig (hunk 4 above)
Comment (yuezhuangshi) [reply .editorconfig]: Done

**Pull request merged**